modules/understanding-pod-identity-webhook-workflow-in-user-defined-projects.adoc

```diff
@@ -0,0 +1,66 @@
+// Module included in the following assemblies:
+//
+// * authentication/managing_cloud_provider_credentials/assuming-an-iam-role-using-the-pod-identity-webhook.adoc
+
+:_content-type: CONCEPT
+[id="understanding-pod-identity-webhook-workflow-in-user-defined-projects_{context}"]
+= Understanding the pod identity webhook workflow in user-defined projects
+
+When you install a {product-title} cluster 
+ifdef::openshift-rosa[]
+that uses the AWS Security Token Service (STS), 
+endif::openshift-rosa[]
+ifndef::openshift-rosa[]
+, 
+endif::openshift-rosa[]
+pod identity webhook resources are included by default.
+
+You can use the pod identity webhook to enable a service account in a user-defined project to assume an AWS Identity and Access Management (IAM) role in a pod in the same project. When the IAM role is assumed, temporary STS credentials are provided for use by the service account in the pod. If the assumed role has the necessary AWS privileges, the service account can run AWS SDK operations in the pod for your AWS account.
+
+To enable the pod identity webhook for a pod, you must create a service account with an `eks.amazonaws.com/role-arn` annotation in your project. The annotation must reference the Amazon Resource Name (ARN) of the AWS IAM role that you want the service account to assume. You must also reference the service account in your `Pod` specification and deploy the pod in the same project as the service account.
+
+[discrete]
+[id="pod-identity-webhook-workflow-in-user-defined-projects_{context}"]
+== Pod identity webhook workflow in user-defined projects
+
+The following diagram illustrates the pod identity webhook workflow in user-defined projects:
+
+.Pod identity webhook workflow in user-defined projects
+image::pod-identity-webhook-workflow-in-user-defined-projects.png[Pod identity webhook workflow]
```

Comment:
In my understanding, in STS mode, there are no more access/secret key on AWS side. Pod uses token (with the specified role on AWS side) to access AWS service. So better to remove these to avoid misleading.
See Figure 1. STS authentication flow at https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-sts.html#cco-mode-sts

Reply:
> The image has some text in light grey at the bottom right (`286_OpenShift_1122`) - not sure if that's intended to show up

@jeana-redhat thank you for highlighting that! I will request an update to the diagram.

Reply:
> In my understanding, in STS mode, there are no more access/secret key on AWS side. Pod uses token (with the specified role on AWS side) to access AWS service. So better to remove these to avoid misleading. See Figure 1. STS authentication flow at https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-sts.html#cco-mode-sts

@jianping-shu thank you for clarifying that. I will remove the access and secret key references from the module and will request an update to the diagram.

Reply:
The diagram has now been updated accordingly.